src/components/Form/EditDiscountForm/EditDiscountForm.tsx

```diff
@@ -322,12 +333,28 @@ const EditDiscountForm = () => {
                     className="px-14 mr-4"
                     color="primary"
                     tag="button"
+                    disabled={
+                      discount.state === DiscountState.Suspended
+                        ? isEqualWith(
+                            initialValues,
+                            values,
+                            (left, right, key) => {
+                              if (key === "productCategories") {
+                                return isEqual(
+                                  [...left].sort(),
+                                  [...right].sort()
+                                );
+                              }
+                            }
+                          )
+                        : false
+                    }
                   >
```

Comment:
For a more readable code, what do you think to move this check in a separated function?

Anyway this check doesn't manage all edge cases, in fact if the user try to edit the percentage amount by delete previous value and typing the same value, the submit button is enabled as showed in the following video:


https://github.com/pagopa/cgn-onboarding-portal-frontend/assets/34343582/39157c7f-4376-4ce4-a5fc-8f31351c62a8



I really suggest you to consider the ad-hoc function addressed in the [comment above](https://github.com/pagopa/cgn-onboarding-portal-frontend/pull/227#pullrequestreview-1539495349)

Reply:
Sure, I can extract Tina separate function

Reply:
If we want to check this specific cases we need to define the behaviour for every field as there are:
- whole strings
- numbers
- trimmed and space normalized strings
- order insensitive arrays

Reply:
If using the mentioned function covers the all the cases I'll use that one.

Reply:
> If we want to check this specific cases we need to define the behaviour for every field as there are:
> 
> * whole strings
> * numbers
> * trimmed and space normalized strings
> * order insensitive arrays

If you consider to compare the `initialValues` with the formik's `values` instead the `discount` state, you don't have to take care about the mentioned cases since the `initialValues` state is already normalized at row 140 (and since that, it follows that also the formik values are already normalized), in fact I suggest you to delete the `discount` state since It is not really needed (at row 302 you could use `initialValues`).

Following this instruction, you can implement the function above without any problem and it will be good for every edge case.

Reply:
Ok

Reply:
The comparation is already between initial values and current values.

Adding trailing spaces in some fields will still enable the save button.

The state field is not part of the form so it will need a type assertion.